File: backend-dotnet/Application/Queries/Posts/GetPostLikesQueryHandler.cs
using Application.DTOs;
using Application.Interfaces;

namespace Application.Queries.Posts
{
    public class GetPostLikesQueryHandler : IQueryHandler<GetPostLikesQuery, PagedResult<SimpleUserDto>>
    {
        private readonly IPostLikeRepository _postLikeRepository;

        public GetPostLikesQueryHandler(IPostLikeRepository postLikeRepository)
        {
            _postLikeRepository = postLikeRepository;
        }

        public async Task<PagedResult<SimpleUserDto>> HandleAsync(GetPostLikesQuery query)
        {
            return await _postLikeRepository.GetPostLikesAsync(query.PostId, query.Page, query.PageSize);
        }
    }
}

File: backend-java/src/main/java/com/jerrygram/application/queries/posts/GetPostLikesQueryHandler.java
package com.jerrygram.application.queries.posts;

import com.jerrygram.application.dtos.PagedResult;
import com.jerrygram.application.dtos.SimpleUserDto;
import com.jerrygram.application.interfaces.IQueryHandler;
import com.jerrygram.infrastructure.repositories.PostLikeRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Slf4j
public class GetPostLikesQueryHandler implements IQueryHandler<GetPostLikesQuery, PagedResult<SimpleUserDto>> {

    private final PostLikeRepository postLikeRepository;

    @Override
    public PagedResult<SimpleUserDto> handle(GetPostLikesQuery query) {
        var pageable = PageRequest.of(query.getPage(), query.getPageSize());
        var users = postLikeRepository.getPostLikesUsers(query.getPostId(), pageable);
        var totalCount = postLikeRepository.countByPostId(query.getPostId());
        
        return PagedResult.<SimpleUserDto>builder()
                .items(users)
                .totalCount((int) totalCount)
                .page(query.getPage())
                .pageSize(query.getPageSize())
                .build();
    }
}